src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/util/SQLiteQueryUtils.java

```diff
@@ -48,24 +48,36 @@
  * A utility class to query cluster, node and resource level summary for a rca
  */
 public class SQLiteQueryUtils {
-  private static final Map<String, String> nestedTableMap;
+  private static final Map<Class<? extends GenericSummary>, List<Class<? extends GenericSummary>>> nestedTableMap;
   private static final Map<String, String> temperatureProfileNestedSummaryMap;
   private static final Set<String> clusterLevelRCA;
   private static final Set<String> temperatureProfileRCASet;
 
-  // to map table => its nested table
-  // e.g. HotClusterSummary => HotNodeSummary
-  static {
-    Map<String, String> tableMap = new HashMap<>();
-    tableMap.put(ClusterTemperatureSummary.TABLE_NAME, ClusterDimensionalSummary.TABLE_NAME);
 
-    tableMap.put(ResourceFlowUnit.RCA_TABLE_NAME, HOT_CLUSTER_SUMMARY_TABLE);
-    tableMap.put(HOT_CLUSTER_SUMMARY_TABLE, HOT_NODE_SUMMARY_TABLE);
-    tableMap.put(HOT_NODE_SUMMARY_TABLE, HOT_RESOURCE_SUMMARY_TABLE);
-    tableMap.put(HOT_RESOURCE_SUMMARY_TABLE, TOP_CONSUMER_SUMMARY_TABLE);
+  // mapping between table => its nested table
+  // RCA API query
+  // |
+  // RcaResponse -- HotClusterSummary -- HotNodeSummary -- HotResourceSummary -- TopConsumerSummary
+  //                                                   |
+  //                                                    -- HotShardSummary
+  static {
+    Map<Class<? extends GenericSummary>, List<Class<? extends GenericSummary>>> tableMap = new HashMap<>();
+    tableMap.put(RcaResponse.class, Collections.unmodifiableList(Collections.singletonList(
+        HotClusterSummary.class)));
+    tableMap.put(HotClusterSummary.class, Collections.unmodifiableList(Collections.singletonList(
+        HotNodeSummary.class)));
+    tableMap.put(HotNodeSummary.class, Collections.unmodifiableList(Collections.singletonList(
+        HotResourceSummary.class)));
+    tableMap.put(HotResourceSummary.class, Collections.unmodifiableList(Collections.singletonList(
+        TopConsumerSummary.class)));
```

Comment:
I think the nesting should be internal to each top level summary rather than in the util class.

Reply:
thanks for pointing this out. yeah this is indeed a much cleaner implementation. I have added the list to each summary class